### DISCUSSION TITLE
Release 0.2.4

### DISCUSSION THREAD

**Discussion by bastianvandenbout** [bastianvandenbout/LISEM#5] [Announcements]
Hi Everyone,

We just release 0.2.4 to the website, which fixes some bugs and adds new features.
There is still a lot to do, and any help is appreciated!

There is also a page on the site describing how to compile lisem on windows (https://lisemmodel.com/docs/compiling/).

Kind regards,
Bastian

**Comment by nariitism** [comment]
Hi Bastian, 
It is great news for all the researchers involved in the hazard modelling and development of such module. I also involved in R&D work related to hazard simulation in 3d view. It will be great pleasure if you let me know the availability of the source code link for your latest release LISEM ver 0.2.4.

Thanking you

Narendra

**Reply by bastianvandenbout** [reply]
Dear Narendra,

Sure no problem, thank you for your interest.
So the source code can be downloaded from this github page.
You can install git and clone https://github.com/bastianvandenbout/LISEM
or download a zip-file of the main branch (0.2.4 at the moment) : https://github.com/bastianvandenbout/LISEM/archive/refs/heads/main.zip 
if you want to compile yourself, see also the small tutorial on the site:
https://lisemmodel.com/docs/compiling/

Of course, feel free to post any issues you encounter here
Kind regards,
Bastian

**Comment by alyrashad** [comment]
Dear Bastian,

I have been experiencing Lisem for a while now, thanks for this amazing software.
Unfortunately, Lisem seems to have a consistency of not working on laptops with a GPU from V 0.2.1 to this version, neither installer nor portable. It works on a laptop that I have with integrated graphics cards. I tried to install it on other laptops with a GPU it always shutdown before it starts. I hope that you have any insight about what solutions might be. It shows me the below message.

[LISEM.txt](https://github.com/bastianvandenbout/LISEM/files/9659160/LISEM.txt)

Best Regards,
Aly Rashad

**Comment by bastianvandenbout** [comment]
Dear Aly,

I see, that is quite unfortunate. I had hoped that 0.2.4 would work for most people even on integrated gpu.
One of the potential causes might be that windows assigns a gpu-device automatically (based on power availability and wether it recognizes the software as something that needs a dedicated gpu).
Lisem often gets assigned the integrated graphics. Then, when the software tries to load OpenCL/OpenGL interoperability for the dedicated gpu, things go wrong.
However, that doesnt exactly match the error output you attached.

One thing you might try is the following:
Within the nvidia control panel there is an option to assign the dedicated gpu to specific software.
Could you set that for LISEM as well
Then, make sure to have the power connected to the laptop and try again.

If that doesnt work, I might compile a special version for you that is a bit more verbose with the output, so we can see exactly where things go wrong.

Kind regards,
Bastian

**Comment by lttrung24** [comment]
Hi Bastian,

Many thanks for your great work. 
I am going to apply Lisem to our research on flashflood and develop a tool based on Python language.
However, after installing the python wheel file of lisem, I got the error "No module named 'lisem.lisem_python'" when importing the package.
I also checked the source code and there is no lisem_python file in the folder.
Could you please check it?

Thank you,
Trung

**Reply by bastianvandenbout** [reply]
Hi Trung,

Happy to hear you find the software usefull!
Very interested to see what you do with it,
For the python stuff, it is more experimental then the software, but all should work fine.
Not all of the funtionality is available (but a lot is, including the flow models).

The python wheel is build using pybind11. It only works on windows and requires python 3.8
I tested it on 3.8.5 and 3.8.9. On 3.7 it might work, but I tested and it doesnt seem to work on pthon 3.9.
I am not sure why Python is so particular about versions and linking with dll libraries.

In your package lisem/ folder (where the wheel is installed), for me that is user/<me>/anaconda3/lib/site-packages/lisem/, there should be a couple of dll files besides python. These link to the actual lisem stuff which is in the subfolder lisem/bin.
If these dlls are missing I have uploaded them seperately. https://sourceforge.net/projects/lisem/files/LISEM%20%28beta%20release%20new%20software%29/python_extra/?upload_just_completed=true 

Anyway, please let me know if it works!

Kind regards,
bastian

**Reply by lttrung24** [reply]
Hi Bastian,

Thank you very much for your prompt reply! 

You are right that the problem is caused by python version and missing of dll files.

After following your instruction, I avoid error "No module named 'lisem.lisem_python'" , however another error "ImportError: DLL load failed while importing lisem_python: A dynamic link library (DLL) initialization routine failed" has been raised.

Could you please let me know how to overcome the problem?
P/s: I am using Windows10 and anaconda python 3.8.5

Kind regards,
Trung

**Reply by bastianvandenbout** [reply]
Hi Trung, sorry for the late reply, some leave this week.
I am not able to easily recreate this.
Perhaps we can have a look together, let me know if you want to chekc it further, my mail is b.vandenbout@utwente.nl

**Reply by georgia-clark** [reply]
Hi Bastian, 

Just came across this thread, I have the same problem with the python installation: "No module named 'lisem.lisem_python'". I have tried with different python version and with and without anaconda but it seems that this module doesn't exist. 

Were you able to resolve this at all / is there an updated .whl file?

Many thanks, 

Georgia

**Reply by skconan** [reply]
Dear Bastian,

I'm encountering the same issue with my Python installation: "No module named 'lisem.lisem_python'." I'm using Anaconda with Python 3.8.5, and I installed LISFLOOD-Single-Event Model (LISEM) from lisem-0.2.4-py3-none-any.whl. 

Could you please provide a solution for this issue? 

Thank you,
SK

**Comment by raigard** [comment]
Dear Bastian, 

I'm studying tools for debris flow modeling, especially those with entrainment equations. I've known LISEM for a while now but haven't used the debris flow yet. Does the LISEM 0.2.4 installer version include the entrainment solutions presented in the paper "Physically based modeling of co-seismic landslide, debris flow, and flood cascade" (https://doi.org/10.5194/nhess-22-3183-2022)? If so, how can I activate this module?

Thank you,
Leonardo Rodolfo Paul

**Comment by 01NancyHuang** [comment]
Dear Bastian,

I hope this email finds you well.I apologize for bothering you. I have conducted some simulation studies using OPENLISEM HAZARD 1.11 and sent you the emails. I am eagerly awaiting your response to the questions and observations I shared in my previous email. Your insights are crucial for my ongoing research, and I believe your guidance will greatly contribute to a better understanding of the model's behavior.Thank you for your time, and I look forward to hearing from you soon.

Best regards.
Nancy Huang

**Comment by nariitism** [comment]
Hello Bastian,

There is no document available for LISEM to download, Only LISEM CLASSIC
document is available for download. Please provide the documention for
LISEM?


Thanks and regards


Dr. Narendra Singh
